cms/teams/tests/test_management_commands.py

```diff
@@ -30,32 +30,51 @@ def api_group(
     }
 
 
-ROLE_GROUP_IDS = {"role-admin", "role-publisher"}
-
-
 @override_settings(
-    ROLE_GROUP_IDS=list(ROLE_GROUP_IDS),
+    SERVICE_AUTH_TOKEN="test-token",
     IDENTITY_API_BASE_URL="https://identity.example",
 )
 class SyncTeamsCommandTests(TestCase):
-    """Covers creation, update, de-activation, dry-run, API/ network errors,
-    and role-group exclusion logic.
+    """Covers creation, update, de-activation, dry-run, API/network errors,
+    auth headers, and role-group exclusion logic.
     """
 
     def _call_command(self, *, dry_run=False):
         argv = ["--dry-run"] if dry_run else []
         with open("/dev/null", "w", encoding="utf-8") as stdout:
             management.call_command("sync_teams", *argv, stdout=stdout)
 
-    def _mock_api(self, groups):
-        """Patch requests.get so /groups returns supplied list."""
+    def assert_command_failure_message(self, cm):
+        self.assertIn("Team sync failed: Failed to fetch groups from the identity API.", str(cm.exception))
+
+    def _mock_api(self, groups, response=None, error=None, expected_headers=None):
```

Comment:
maybe one for the follow auth ticket, but we have `responses` and it would be good to switch to it https://github.com/getsentry/responses/blob/master/README.rst

example usage: https://github.com/ONSdigital/dis-wagtail/blob/dca8fe7cd67e798b8cc807def36f77021bba54d7/cms/datasets/tests/test_models.py#L16-L31

Reply:
I was debating this, and I'm all in favour of using `responses` 🎉 . I want to keep the changes to a minimum in this PR. Let's follow up.